### ISSUE TITLE
Support enum aliases in `Literal` type checking

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#3077]
For the assignment in the following code Pyright currently reports a type error:

```python
import enum
from typing import Literal

class Foo(enum.Enum):
    A = 1
    B = A

_: Literal[Foo.A] = Foo.B
```

It would be cool if Pyright could understand enum aliases.

**Comment by erictraut** [comment]
I'm not sure if this should type check. `Foo.A` and `Foo.B` are distinct symbols, so `Literal[A]` and `Literal[B]` are different types. This is consistent with mypy's behavior as well.

**Comment by erictraut** [comment]
Pyright and mypy already include a ridiculous amount of special casing to handle the basic functionality of the `Enum` class, which is one of the most over-engineered and unnecessarily complex constructs that I've ever seen in any standard library. I'd prefer not to add additional special cases to handle obscure edge cases like this one.

I'm going to close this request for now. We might reconsider the request in the future if we receive signal from enough other pyright users that this is something they want to see.
